exonum-java-binding/core/src/main/java/com/exonum/binding/core/storage/database/MemoryDb.java

```diff
@@ -71,11 +72,15 @@ public Fork createFork(Cleaner cleaner) {
 
   /**
    * Applies the changes from the given fork to the database state.
+   * Invalidates provided fork, as it can't be used after merge.
    *
    * @param fork a fork to get changes from
    */
   public void merge(Fork fork) {
-    nativeMerge(getNativeHandle(), fork.getViewNativeHandle());
+    NativeHandle forkNativeHandle = fork.getViewNativeHandle();
```

Comment:
I think it would be better if the fork itself performed the operation of converting itself
into a patch.

The problems with the code:
  - It frees the fork handle using unrelated free method for the MemoryDb, which will likely crash.
  - There is unclear interference with the Fork Cleaner.
  - The indices created with this fork remain live:
     - They are not destroyed before Fork (they will be after, which might cause a crash)
     - They are not invalidated — allowing subsequent operations (which will definitely cause a crash).
  - I think each type must free its own handles, and destroy other proxies through their public API.